src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -109,6 +115,13 @@ protected virtual bool CanUpdateProperty([NotNull] ModelMetadata propertyMetadat
 
         private async Task<bool> CanValueBindAnyModelProperties(MutableObjectBinderContext context)
         {
+            // If there are no properties on the model, there is nothing to bind. We are here means this is not a top 
```

Comment:
This statement seems incorrect - wouldn't we be here if we are a top level object with some properties? Also, could we rewrite the earlier condition as

```C#
if (context.PropertyMetadata != null && context.PropertyMetadata.Count == 0)
{
   return isTopLevelObject;
}
```

We wouldn't require this check here.